app/script/view_model/content/InputBarViewModel.js

```diff
@@ -217,8 +215,30 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
   }
 
   addMention(userEntity) {
-    // TODO
+    const editedMention = this.editedMention();
+    const mentionData = {
+      length: userEntity.name().length + 1,
+      start: editedMention.start,
+      user_id: userEntity.id,
+    };
+    const mention = new z.message.MentionEntity(mentionData);
```

Comment:
MentionEntity take a proto mention in the constructor. For this the proper approach would be

````
const mentionEntity = new z.message.MentionEntity().setUserIdMention(this.editedMention().start, userEntity.name().length + 1, userEntity.id);
````